src/foundation.cs

```diff
@@ -16594,4 +16594,50 @@ enum NSUrlErrorNetworkUnavailableReason : long {
 		Expensive = 1,
 		Constrained = 2,
 	}
+
+	[NoWatch, NoTV, NoiOS, Mac (10,10)]
+	[Native]
+	public enum NSBackgroundActivityResult : long
+	{
+		Finished = 1,
+		Deferred = 2,
+	}
+
+	delegate void NSBackgroundActivityCompletionHandler (NSBackgroundActivityResult result);
+
+	delegate void NSBackgroundActivityCompletionAction ([BlockCallback] NSBackgroundActivityCompletionHandler handler);
+
+	[NoWatch, NoTV, NoiOS, Mac (10,10)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface NSBackgroundActivityScheduler
+	{
+		[Export ("initWithIdentifier:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (string identifier);
+
+		[Export ("identifier")]
+		string Identifier { get; }
+
+		[Export ("qualityOfService", ArgumentSemantic.Assign)]
+		NSQualityOfService QualityOfService { get; set; }
+
+		[Export ("repeats")]
+		bool Repeats { get; set; }
+
+		[Export ("interval")]
+		double Interval { get; set; }
+
+		[Export ("tolerance")]
+		double Tolerance { get; set; }
+
+		[Export ("scheduleWithBlock:")]
+		void Schedule (NSBackgroundActivityCompletionAction action);
```

Comment:
During team week @dalexsoto mentioned the `Async` attribute should be used when the last argument of a method is a callback. Why isn't async used here?

Reply:
@wachaudh because in this case, it is not a function to be called when the method is completed with the result of the operation, but a function to be schedule to be called. 

That is, if the callback taken by the function is going to be called with the results of the function, you should use Async. In this case you are just giving a function to be schedule.